scalding-core/src/main/scala/com/twitter/scalding/serialization/PositionInputStream.scala

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import java.io.InputStream
+import JavaStreamEnrichments._
+
+object PositionInputStream {
+  def apply(in: InputStream): PositionInputStream = in match {
+    case p: PositionInputStream => p
+    case nonPos => new PositionInputStream(nonPos)
+  }
+}
+
+class PositionInputStream(val wraps: InputStream) extends InputStream {
+  private[this] var pos: Long = 0L
+  private[this] var markPos: Long = -1L
+  def position: Long = pos
+
+  override def available = wraps.available
+
+  override def close() { wraps.close() }
+
+  override def mark(limit: Int) {
+    wraps.mark(limit)
+    markPos = pos
+  }
+
+  override val markSupported: Boolean = wraps.markSupported
+
+  override def read: Int = {
+    val result = wraps.read
+    // returns -1 on eof, otherwise non-negative number
+    if (result >= 0) pos += 1
+    result
+  }
+  override def read(bytes: Array[Byte]): Int = read(bytes, 0, bytes.length)
```

Comment:
can't imagine that it matters, but the abstract class InputStream defaults to delegating this method to read(bytes, off, len) but allows it to be overridden, which means some wacky input stream *might* handle read(bytes) differently than read(bytes, 0, bytes.length) (can't imagine why tho) but here you're not delegating to that method.